components/CollectioncardForm.js

```diff
@@ -0,0 +1,79 @@
+import styled from "styled-components";
+import Button from "./Button";
+
+const StyledForm = styled.form`
+  position: relative;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  gap: 8px;
+  margin: 4px 24px 4px 24px;
+  padding: 0 0 16px 0;
+`;
+
+const Label = styled.label`
+  align-self: flex-start;
+  font-size: 16px;
+  font-weight: 600;
+  color: ${({ theme }) => theme.modalText};
+`;
+
+const Input = styled.input`
+  width: 100%;
+  padding: 8px;
+  border: 1px solid #ccc;
+  border-radius: 4px;
+  margin-bottom: 16px;
+`;
+
+const ButtonContainer = styled.div`
+  display: flex;
+  justify-content: space-between;
+  width: 70%;
+  padding: 32px 0 8px 0;
+`;
+
+export default function CollectioncardForm({
+  onSubmit,
+  initialValues,
+  onClose,
+}) {
+  async function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+
+    const mergedData = {
+      ...initialValues,
+      ...data,
+      isCorrect: initialValues ? initialValues.isCorrect : false,
+    };
+
+    if (initialValues?._id) {
+      await onSubmit(initialValues._id, mergedData);
+    } else {
+      await onSubmit(mergedData);
+    }
+  }
```

Comment:
Seems like you added the same code as from your `FlashcardForm` here. But as you want to update a collection here, you only need to save the `_id` and `title` from the collection.
As mentioned in a previous comment, you don't need to pass the `_id` separately as it is already included in `data`.

```suggestion
    const mergedData = {
      ...initialValues,
      ...data,
    };
      await onSubmit(mergedData);
  }
```